challenges/largest_product_array/largest_product.py

```diff
@@ -0,0 +1,14 @@
+def largest_product(arr):
+    """
+    Write a function which takes a 2D array and returns the largest product of
+    2 adjacent values.
+    """
+    if not arr:
+        return 'empty arr'
+    largest_prod = 0
+    product = 0
+    for index in arr:
+        product = index[0] * index[1]
+        if product > largest_prod:
+            largest_prod = product
+    return largest_prod
```

Comment:
This only multiplies numbers next to each other horizontally. You also need to check vertically as well.
```
[[1, 3], [6, 10], [4, 5]]
```
represents the matrix
```
1  3
6  10
4  5
```
Thus you need to multiply `1 * 3` as well as `1 * 6`.
You should also make this work for any size matrix, not just a 2xn.